feat: ✨ add sd-table and sd-table-cell

MarcMatthiae opened the pull request:
## Description:
Add css components sd-table and sd-table-cell.

## Definition of Reviewable:
- [x] Documentation is created/updated
- [x] Migration Guide is created/updated
- [x] Stories (features, a11y) are created/updated
- [x] relevant tickets are linked
- [x] PR is assigned to project board

karlbaumhauer commented on packages/components/src/styles/table-cell/table-cell.stories.ts:
```diff
@@ -0,0 +1,275 @@
+  }
+};
+
+// TODO: Add sample "Simple Table, First Column Fixed" and "Multi Select Table" as soon as we have sd-scrollable.
```
pls remove the ToDo and open up a ticket for this. ToDos in the code usually get lost... ;-)

MarcMatthiae replied:
#376

mariohamann commented:
<img width="1034" alt="CleanShot 2023-09-13 at 12 40 44@2x" src="https://github.com/solid-design-system/solid/assets/26542182/24106398-b9ac-40c5-a01d-b1a00c0d9293">

Please make the background colors an enum, calling it `sd-table-cell--bg-...` to make clear, that this is only for the background.

See e. g. 
<img width="1045" alt="CleanShot 2023-09-13 at 12 42 34@2x" src="https://github.com/solid-design-system/solid/assets/26542182/dbdc0317-0924-411a-8bc3-e64d3fab5255">

mariohamann commented on packages/components/src/styles/table-cell/table-cell.stories.ts:
```diff
@@ -0,0 +1,275 @@
+          padding: 0;
+        }
+
+        .sd-table-cell.sortable button {
+          display: inline-flex;
+          box-sizing: border-box;
+          height: 100%;
+          width: 100%;
+          margin: 0;
+          padding: 16px;
+          align-items: center;
+          justify-content: flex-start;
+          gap: 16px;
+          text-align: left;
+        }
+
+        .sd-table-cell.sortable button:hover {
+          background-color: white;
```
It would be helpful to make this visible in the story's code.

yoezlem replied:
Maybe we can also replace the CSS part with our tokens...?

MarcMatthiae replied:
I wanted to do that, but I could not find out how to use our tokens in storybook. Do you know how to do that?

MarcMatthiae commented:
> <img alt="CleanShot 2023-09-13 at 12 40 44@2x" width="1034" src="https://user-images.githubusercontent.com/26542182/267625084-24106398-b9ac-40c5-a01d-b1a00c0d9293.png">
> Please make the background colors an enum, calling it `sd-table-cell--bg-...` to make clear, that this is only for the background.
> 
> See e. g. <img alt="CleanShot 2023-09-13 at 12 42 34@2x" width="1045" src="https://user-images.githubusercontent.com/26542182/267625430-dbdc0317-0924-411a-8bc3-e64d3fab5255.png">

I understood, that we provide this as part of the table-cell component and we also wrote that in the story: 
![image](https://github.com/solid-design-system/solid/assets/3873797/631ae6c1-8026-4e43-908a-3ebc959246d5)

I don´t know if it is necessary, but at least we refined it like that.

mariohamann commented:
> > <img alt="CleanShot 2023-09-13 at 12 40 44@2x" width="1034" src="https://user-images.githubusercontent.com/26542182/267625084-24106398-b9ac-40c5-a01d-b1a00c0d9293.png">
> > Please make the background colors an enum, calling it `sd-table-cell--bg-...` to make clear, that this is only for the background.
> > See e. g. <img alt="CleanShot 2023-09-13 at 12 42 34@2x" width="1045" src="https://user-images.githubusercontent.com/26542182/267625430-dbdc0317-0924-411a-8bc3-e64d3fab5255.png">
> 
> I understood, that we provide this as part of the table-cell component and we also wrote that in the story: ![image](https://user-images.githubusercontent.com/3873797/267887338-631ae6c1-8026-4e43-908a-3ebc959246d5.png)
> 
> I don´t know if it is necessary, but at least we refined it like that.

Yeah, but we should document it properly. Just provide it as an own style. :)

mariohamann commented on packages/components/src/styles/table-cell/table-cell.css:
```diff
@@ -0,0 +1,37 @@
+  }
+}
+
+.sd-table-cell * {
+  @apply align-middle;
+}
```
But maybe it should. :)

MarcMatthiae replied:
Actually in my samples this did not make a difference, so I can just remove it.

yoezlem commented on packages/components/src/styles/table-cell/table-cell.css:
```diff
@@ -0,0 +1,37 @@
+*/
+.sd-table {
+  all: unset;
+  display: table;
```
Why did you not add the tailwind classes?

@apply table border-collapse border-spacing-0



<br class="Apple-interchange-newline">

MarcMatthiae replied:
Good question...Will change this.

MarcMatthiae replied:
Moved to new sd-table component and updated to tw classes.

MarcMatthiae commented:
> > > <img alt="CleanShot 2023-09-13 at 12 40 44@2x" width="1034" src="https://user-images.githubusercontent.com/26542182/267625084-24106398-b9ac-40c5-a01d-b1a00c0d9293.png">
> > > Please make the background colors an enum, calling it `sd-table-cell--bg-...` to make clear, that this is only for the background.
> > > See e. g. <img alt="CleanShot 2023-09-13 at 12 42 34@2x" width="1045" src="https://user-images.githubusercontent.com/26542182/267625430-dbdc0317-0924-411a-8bc3-e64d3fab5255.png">
> > 
> > 
> > I understood, that we provide this as part of the table-cell component and we also wrote that in the story: ![image](https://user-images.githubusercontent.com/3873797/267887338-631ae6c1-8026-4e43-908a-3ebc959246d5.png)
> > I don´t know if it is necessary, but at least we refined it like that.
> 
> Yeah, but we should document it properly. Just provide it as an own style. :)

Done :)

MarcMatthiae commented on packages/components/src/styles/table-cell/table-cell.stories.ts:
```diff
@@ -0,0 +1,275 @@
+    );
+
+    // Function to exchange the sort icons and set the aria-sort attribute in the sortable table after a click on the table header
+    const sortTable = (column: number) => {
```
Done :)

coraliefeil commented:
Review done

MarcMatthiae commented:
Main changes
- Removed "Sortable Table" sample
- Added "Simple Table" sample
- Added correct font color to sd-table-cell
- @coraliefeil Leider brauchen wir diese "default"-Farbe immer, haben wir am Freitag im Daily besprchen. Ich habe jetzt zumindest eingebaut, dass die Farbe in Storybook erstmal auf transparent spring, aber es gibt immer diesen (Default)-Eintrag.

mariohamann commented on packages/components/src/styles/table-cell/table-cell.css:
```diff
@@ -0,0 +1,25 @@
+  }
+
+  &--bg {
+    &-transparent {
+      @apply bg-transparent;
+    }
```
I think this should be part the default. Please remove it here and in the declaration and add bg-transparent to line 2.

mariohamann commented on packages/components/src/styles/table-cell/table-cell.declaration.ts:
```diff
@@ -0,0 +1,19 @@
+    {
+      name: 'sd-table-cell--bg-...',
+      options: ['transparent', 'white', 'primary-100', 'neutral-100'],
+      description: 'Applies the selected backround-color to table cell.'
```
```suggestion
      description: 'Applies the selected background-color to table cell.'
```
